docs/getting-started/serving-for-inference-and-training.md

```diff
@@ -12,6 +12,27 @@ To serve a feature we initialize a serving client, specify the names and variant
 fpf = client.features([("fpf", "quickstart")], {"passenger": "1"})
 ```
 
+### On-Demand Features
+
+On-demand features do not require any materialization, they are calculated at serving time. You can define your on-demand features by simply adding the `ondemand_feature` decorator to your function. The function will need to have `serving_client`, `params`, `entities`, as part of it's definition.
+
+```python
+import featureform as ff
+
+@ff.ondemand_feature(name="sex_as_int", variant="quickstart")
+def sex_as_int(serving_client, params, entities):
+    passenger_id = params[0]
+    df = params[1]
+    df['Sex'] = df['Sex'].map( {'female': 0, 'male': 1} ).astype(int)
+    return df.loc[df["PassengerId"] == passenger_id]["Sex"].values
```

Comment:
Does this return a list?

Reply:
it would be numpy.ndarray of size 1.

Reply:
It should just return a scalar right?

Reply:
it works fine as is but I will change it to scalar